src/net/sourceforge/kolmafia/utilities/ResettingHttpClient.java

```diff
@@ -29,34 +27,16 @@ public ResettingHttpClient(Supplier<HttpClient> createClient) {
   }
 
   public void resetClient() {
-    closeClient(client);
-
     this.client = createClient.get();
     clientRequestsSent.set(0);
   }
 
   public <T> HttpResponse<T> send(HttpRequest req, HttpResponse.BodyHandler<T> handler)
       throws IOException, InterruptedException {
+    var resp = this.client.send(req, handler);
     if (clientRequestsSent.incrementAndGet() >= HTTP_CLIENT_REQUEST_LIMIT) {
       resetClient();
     }
-
-    return this.client.send(req, handler);
-  }
-
-  private void closeClient(HttpClient httpClient) {
-    // Closing the client is a new java feature
-    if (Runtime.version().feature() < 21) {
-      return;
-    }
-
-    // As we compile against an older jdk, we must use reflection to access the method
-    try {
-      Method method = Class.forName("java.net.http.HttpClient").getMethod("close");
-      method.invoke(httpClient);
-    } catch (Exception ex) {
-      // If this does error, we should log it as this means something has changed in java internals
-      StaticEntity.printStackTrace(ex);
-    }
+    return resp;
```

Comment:
can you put the change
```
return this.client.send(req, handler);
```
in this PR as well? I think this is a correctness fix: if not, multiple threads can send requests before the request limit is checked. That isn't Java 21 specific.